lib/proxy/events/log_decorator.rb

```diff
@@ -0,0 +1,59 @@
+require 'proxy/events/buffer'
+
+module Proxy::Events
+  class LogDecorator
+    def initialize(logger)
+      @logger = logger
+    end
+
+    def add(severity, message = nil, progname = nil, backtrace = nil, &block)
```

Comment:
What do you mean? It is not used in smart-proxy (the block syntax) but it is the logger contract and I don't want to break it. Logging should be working "as is".